src/components/cartPage/howDiscoverPicker.jsx

```diff
@@ -58,79 +57,72 @@ function HowDiscoverPicker({
-              )}
-              <button
-                id="picker-param__status-more"
-                type="button"
-                aria-label="Mute volume"
-                onClick={() => {
-                  setOpened(!isOpened);
-                }}
-                data-cy="howDiscover-picker"
-              />
-            </div>
-            <div className="line" />
-            <ul className="picker-param__status-list">
-              {networksList.map((item) => (
-                <li
-                  onClick={handleSelect.bind(null, item)}
-                  key={item.name}
-                  className="list-elem"
-                >
-                  <span className="param__status-list__title">
-                    {lang(item.name, language)}
-                  </span>
-                </li>
-              ))}
-            </ul>
+
+        <div
+          className={
+            error && touched
+              ? 'picker-param__status error'
+              : 'picker-param__status'
+          }
+          ref={ref}
+        >
+          <div className="picker-param__status-main">
+            {pickedItem?.name === 'Other' && (
+              <>
+                <Field
+                  name="howDiscover"
+                  id="howDiscover"
+                  type="text"
+                  value={value}
+                  onChange={handleChange}
+                  className="how-discover-option"
+                  placeholder={lang('EnterYourOwnOption', language)}
+                />
+              </>
+            )}
+            {pickedItem && pickedItem.name !== 'Other' && (
+              <>
+                <Field
+                  name="howDiscover"
+                  id="howDiscover"
+                  type="text"
+                  value={lang(pickedItem.name, language)}
+                  className="how-discover-option"
+                />
+              </>
+            )}
+            {!pickedItem && (
+              <>
+                <span className="param__status-choose__title">
+                  {lang('SelectFromList', language)}
+                </span>
+              </>
+            )}
+            <button
+              id="picker-param__status-more"
+              type="button"
+              aria-label="Mute volume"
+              onClick={() => {
+                setOpened(!isOpened);
+              }}
+              data-cy="howDiscover-picker"
+            />
           </div>
-        )}
-        {/* </FocusWithin> */}
+          <div className="line" />
+          <ul className="picker-param__status-list">
+            {networksList.map((item) => (
+              <li
+                onClick={handleSelect.bind(null, item)}
+                key={item.name}
+                className="list-elem"
+              >
+                <span className="param__status-list__title">
+                  {lang(item.name, language)}
+                </span>
+              </li>
+            ))}
+          </ul>
```

Comment:
I'll skip it for now.

Reply:
No, thank you, I decided to add accessibility, because SonarCloud requires to have a better mark for it - [d924eb5](https://github.com/boarlabsxyz/Kavoon/pull/5/commits/d924eb50bfb0c16e6e4639da6e31481692b0ad73), thanks.